modos/api.py

```diff
@@ -451,44 +451,40 @@ def from_file(
         htsget_endpoint: Optional[str] = None,
     ) -> MODO:
         """build a modo from a yaml or json file"""
-        instances = parse_multiple_instances(Path(path))
-        # check for unique ids and fail early
-        ids = [inst.id for inst in instances]
-        if len(ids) > len(set(ids)):
-            dup = {x for x in ids if ids.count(x) > 1}
-            raise ValueError(
-                f"Please specify a unique ID. Element(s) with ID(s) {dup} already exist."
-            )
-        # use full id for has_part attributes
-        instances = [update_haspart_id(inst) for inst in instances]
-
-        modo_inst = [
-            instance
-            for instance in instances
-            if isinstance(instance, model.MODO)
-        ]
-        if len(modo_inst) != 1:
+        element_list = parse_attributes(Path(path))
+        modo_count = len(
+            [
+                ele
+                for ele in element_list
+                if ele["metadata"].get("@type") == "MODO"
+            ]
+        )
+        if modo_count != 1:
```

Comment:
I don't really like this code, but I thought it could be a useful check that should not be removed, just to keep the code cleaner. Happy about any suggestion?

Reply:
I agree it's good to check it, and the code is fine IMO. A slightly shorter variant would be this, but this is basically the same.

```python
sum(
    [
        ele["metadata"].get("@type") == "MODO"
        for ele in element_list
    ]
)
```

or in functional style:
```python
sum(
    map(
        lambda e: e['metadata'].get('@type') == 'MODO',
        element_list
    )
)
```